### PR TITLE
fixes #13010 - properly calculate products for a system to include re…

### PR DESCRIPTION
**Pull request by jlsherrill**
…dhat

### CODE REVIEW

**Comment by jlsherrill** [comment]
@johnpmitsch this is kinda related to something you worked on, care to review?

File: app/models/katello/glue/candlepin/consumer.rb [hunk 1]
```diff
@@ -405,16 +405,8 @@ def populate_from(candlepin_systems)
...
-        self.entitlements.each do |entitlement|
-          pool = Katello::Pool.find_by(:cp_id => entitlement['pool']['id'])
-          Katello::Product.where(:cp_id => pool.product_id).each do |product|
-            all_products << product
-          end
-        end
-
-        return all_products
+        pool_ids = self.entitlements.map { |entitlement| entitlement['pool']['id'] }
+        Katello::Product.joins(:subscriptions => :pools).where("#{Katello::Pool.table_name}.cp_id" => pool_ids).enabled.uniq
```
Comment (johnpmitsch) [inline app/models/katello/glue/candlepin/consumer.rb]: @jlsherrill just curious, what does ```.enabled``` do?

**Comment by johnpmitsch** [comment]
@jlsherrill everything LGTM

File: app/models/katello/glue/candlepin/consumer.rb (hunk 1 above)
Comment (johnpmitsch) [inline app/models/katello/glue/candlepin/consumer.rb]: @jlsherrill ah, nvm, [found it](https://github.com/Katello/katello/blob/master/app/models/katello/product.rb#L59-L62)

**Comment by johnpmitsch** [comment]
ACK

**Comment by jlsherrill** [comment]
Thanks for the review @johnpmitsch !

**Pull request merged**